OpenRA.Mods.Common/Traits/World/MPStartLocations.cs

```diff
@@ -18,24 +18,58 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
-	public class MPStartLocationsInfo : ITraitInfo
+	[Desc("Controls the 'Separate Team Spawns' checkbox in the lobby options.")]
+	public class MPStartLocationsInfo : ITraitInfo, ILobbyOptions
 	{
 		public readonly WDist InitialExploreRange = WDist.FromCells(5);
 
+		[Translate]
+		[Desc("Descriptive label for the spawn positions checkbox in the lobby.")]
+		public readonly string CheckboxLabel = "Separate Team Spawns";
+
+		[Translate]
+		[Desc("Tooltip description for the spawn positions checkbox in the lobby.")]
+		public readonly string CheckboxDescription = "Players without assigned spawn points will start as far as possible from enemy players.";
+
+		[Desc("Default value of the spawn positions checkbox in the lobby.")]
+		public readonly bool CheckboxEnabled = true;
+
+		[Desc("Prevent the spawn positions state from being changed in the lobby.")]
+		public readonly bool CheckboxLocked = false;
+
+		[Desc("Whether to display the spawn positions checkbox in the lobby.")]
+		public readonly bool CheckboxVisible = true;
+
+		[Desc("Display order for the spawn positions checkbox in the lobby.")]
+		public readonly int CheckboxDisplayOrder = 7;
+
 		public virtual object Create(ActorInitializer init) { return new MPStartLocations(this); }
+
+		IEnumerable<LobbyOption> ILobbyOptions.LobbyOptions(Ruleset rules)
+		{
+			yield return new LobbyBooleanOption("randomspawnpositions", CheckboxLabel, CheckboxDescription, CheckboxVisible, CheckboxDisplayOrder, CheckboxEnabled, CheckboxLocked);
+		}
 	}
 
-	public class MPStartLocations : IWorldLoaded
+	public class MPStartLocations : IWorldLoaded, INotifyCreated
 	{
 		readonly MPStartLocationsInfo info;
 
 		public readonly Dictionary<Player, CPos> Start = new Dictionary<Player, CPos>();
 
+		public static bool SeparateTeamSpawns { get; private set; }
```

Comment:
Please define this as `bool separateTeamSpawns;`.  It's not used by outside code, and shouldn't be static!